src/core/execute.c

```diff
@@ -1719,7 +1719,15 @@ static int exec_child(
                 }
 #ifdef SMACK_DEFAULT_PROCESS_LABEL
                 else {
-                        r = mac_smack_apply_pid(0, SMACK_DEFAULT_PROCESS_LABEL);
+                        _cleanup_free_ const char *exec_label = NULL;
```

Comment:
This is broken. A pointer passed to `free()` cannot be `const`. gcc does not warn you about this because of our casting logic. We have ~3 other occurrences other places in our tree which need fixing.

Your `const char **label` arguments of the functions above need to drop the `const` assignment hence.